src/authStrategies/RemoteAuth.js

```diff
@@ -0,0 +1,221 @@
+const path = require('path');
+const unzipper = require('unzipper');
+const archiver = require('archiver');
+const BaseAuthStrategy = require('./BaseAuthStrategy');
+
+/**
+ * Remote-based authentication
+ * @param {object} options - options
+ * @param {object} options.store - Remote database store instance
+ * @param {string} options.clientId - Client id to distinguish instances if you are using multiple, otherwise keep null if you are using only one instance
+ * @param {string} options.dataPath - Change the default path for saving session files, default is: "./.wwebjs_auth/" 
+ * @param {number} options.backupSyncMs - Sets the time interval for periodic session backups. Accepts values starting from 60000ms {1 minute}
+ */
+class RemoteAuth extends BaseAuthStrategy {
+    constructor({ clientId, dataPath, store, backupSyncMs } = {}) {
+        super();
+
+        const idRegex = /^[-_\w]+$/i;
+        if (clientId && !idRegex.test(clientId)) {
+            throw new Error('Invalid clientId. Only alphanumeric characters, underscores and hyphens are allowed.');
+        }
+        if (!backupSyncMs || backupSyncMs < 60000) {
+            throw new Error('Invalid backupSyncMs. Accepts values starting from 60000ms {1 minute}.');
+        }
+        if(!store) throw new Error('Remote database store is required.');
+
+        this.store = store;
+        this.clientId = clientId;
+        this.backupSyncMs = backupSyncMs;
+        this.dataPath = path.resolve(dataPath || './.wwebjs_auth/');
+        this.requiredDirs = ['Default', 'IndexedDB', 'Local Storage']; /* => Required Files & Dirs in WWebJS to restore session */
+    }
+
+    async beforeBrowserInitialized() {
+        const puppeteerOpts = this.client.options.puppeteer;
+        const sessionDirName = this.clientId ? `session-${this.clientId}` : 'session';
+        const dirPath = path.join(this.dataPath, sessionDirName);
+
+        if (puppeteerOpts.userDataDir && puppeteerOpts.userDataDir !== dirPath) {
+            throw new Error('RemoteAuth is not compatible with a user-supplied userDataDir.');
+        }
+
+        this.userDataDir = dirPath;
+        this.sessionName = sessionDirName;
+
+        await this.extractRemoteSession();
+
+        this.client.options.puppeteer = {
+            ...puppeteerOpts,
+            userDataDir: dirPath
+        };
+    }
+
+    async logout() {
+        await this.deleteRemoteSession();
+
+        let pathExists = await this.isValidPath(this.userDataDir);
+        if (pathExists) {
+            await fs.promises.rm(this.userDataDir, {
+                recursive: true,
+                force: true
+            }).catch(() => {});
+        }
+        clearInterval(this.backupSync);
+    }
+
+    async authReady() {
+        const sessionExists = await this.store.sessionExists({session: this.sessionName});
+        if(!sessionExists) {
+            await this.delay(30000); /* Initial delay sync required for session to be stable enough to recover */
```

Comment:
Does this mean that if you close the client before this initial delay, you will start back up with a fresh new session and have to authenticate again? Is there something other than a fixed delay that we can use to know if it's "restorable" (e.g. some file present or something set on the client)?

Reply:
Yes, this means exactly that. To be honest, I don't know at a granular level what files are the ones being updated for the session to become stable, but as far as I went with my research, I was able to detect that a fixed 30 sec delay before the initial saving was all we needed.

Reply:
This may be different based on the computer's resources. Especially from my experiments in AWS lambda I saw that many operations took half the time with doubling the memory and this may also be affected

Reply:
I see.. suggestions and improvements are more than welcome as this is mainly based on my own research & personal experience on the matter.